### PR TITLE
Pipe

### PR DESCRIPTION
**Pull request by dotnetCarpenter**
This is my initial PR for reading piped reports - #29 . It needs a little more work and I would like to hear your opinion.

3 questions:
1. What should the [path be for piped reports](https://github.com/codecov/codecov-node/compare/master...dotnetCarpenter:pipe?expand=1#diff-6a66219cd60d8f848159130aa118c38bR253)?
2. [The check for `if (files) ...`](https://github.com/codecov/codecov-node/compare/master...dotnetCarpenter:pipe?expand=1#diff-6a66219cd60d8f848159130aa118c38bR286) will always be true. Why not remove it?
3. ~~What is the difference between lcov and coverage.json, except the data format? E.g. when is one generated over the other and is it by `codecov-node` or an 3th party?~~ - just saw your answer in [comment-244543227](https://github.com/tapjs/node-tap/issues/298#issuecomment-244543227)

### CODE REVIEW

**Comment by dotnetCarpenter** [comment]
@stevepeak  For some reason the test goes stale after passing all tests. Not sure what is going on yet.

**Comment by dotnetCarpenter** [comment]
@stevepeak finally it passes. But codecov doesn't seem to pick it up?

File: bin/codecov [hunk 1]
```diff
@@ -21,4 +21,25 @@ var args = argv.option([
...
+  }
+});
+process.stdin.on('end', function() {
+  if (readingPiped) codecov.upload(args);
+});
+
+var timer = setTimeout(function() {
+  if (!readingPiped) {
+    codecov.upload(args);
+    process.exit();
```
Comment (stevepeak) [inline bin/codecov]: I think this may be exiting before uploading is complete.

**Comment by stevepeak** [comment]
The process seems to quite before uploading is started. You can see this in the build logs. I think the issue is beacuse of the pipe timeout/exit.

**Comment by eddiemoore** [comment]
Closing due to inactivity

**Comment by dotnetCarpenter** [comment]
I'll revisit this in January.

**Comment by hadrienk** [comment]
It would be just fine without the timeout IMO. Most CI services are getting slow depending on global usage. A timeout will only make the report fails inconsistently.

Let me know if you don't have time to work on this I could try to take over.

**Comment by dotnetCarpenter** [comment]
@hadrienk thanks for taking over. I just don't have time ATM

**Comment by dotnetCarpenter** [comment]
@hadrienk I vaguely remember having the timeout to distinguish between piped input and file input but I don't remember if I actually tested this or if it's the remnants of an issue with a different project I wrote, that has the same listening implementation.

**Comment by dotnetCarpenter** [comment]
So I just need to remove the timeout? I should be able to find time for that soon.

**Comment by dotnetCarpenter** [comment]
@hadrienk Checkout #92

**Comment by hadrienk** [comment]
Thanks! I will test it tomorrow.

**Pull request closed without merging**